app/main.py

```diff
@@ -1,2 +1,56 @@
+from typing import Any
+
+
 class Dictionary:
-    pass
+
+    class Node:
+        def __init__(self, key: Any, value: Any) -> None:
+            self.key = key
+            self.value = value
+            self.hash = hash(key)
+
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.size = 0
+        self.load_factor = 2 / 3
+        self.hash_table = [[] for _ in range(self.capacity)]
+
+    def __len__(self) -> int:
+        return self.size
+
+    def index(self, key: Any) -> int:
+        return hash(key) % self.capacity
+
+    def resize(self) -> None:
+        new_capacity = self.capacity * 2
+        new_hash_table = [[] for _ in range(new_capacity)]
+
+        for bucket in self.hash_table:
+            for node in bucket:
+                index = node.hash % new_capacity
+                new_hash_table[index].append(node)
+
+        self.hash_table = new_hash_table
+        self.capacity = new_capacity
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        if self.size / self.capacity > self.load_factor:
```

Comment:
The condition `self.size / self.capacity > self.load_factor` uses integer division, which will always result in zero since both `self.size` and `self.capacity` are integers. This means the resize method will never be called. You should use float division instead by converting one of the operands to a float, for example: `self.size / float(self.capacity) > self.load_factor`.